### PR TITLE
[SRVKE-1076] Run eventing-kafka-broker e2e tests in SO

### PR DESCRIPTION
**Pull request by pierDipi**
We will run eventing-kafka-broker tests in a separate
job since it takes too long to run all upstream tests in
one single job, see https://github.com/openshift-knative/serverless-operator/pull/1731

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

Fixes JIRA https://issues.redhat.com/browse/SRVKE-1076

## Proposed Changes

- Run ekb upstream tests using a separate Makefile target `test-upstream-e2e-kafka-no-upgrade` (or the variant -testonly)

### CODE REVIEW

**Comment by matzew** [comment]
FWIW, I just rebased this PR

File: test/eventing-kafka.bash [hunk 1]
```diff
@@ -29,3 +29,29 @@ function upstream_knative_eventing_kafka_e2e {
...
 }
+
+function upstream_knative_eventing_kafka_broker_e2e {
+  logger.info 'Setting Kafka as default broker class'
+
+  local root_dir
+  root_dir=$(dirname "$0")/..
+
+  # Set Kafka Broker as default Broker class
+  oc patch knativeeventing -n knative-eventing knative-eventing --patch-file "${root_dir}/test/config/eventing/kafka-broker-default-patch.yaml"
```
Comment (pierDipi) [inline test/eventing-kafka.bash:40]: my bash foo isn't working but I don't know what wrong :thinking: 
```
error: unable to read patch file: open ./test/../test/config/eventing/kafka-broker-default-patch.yaml: no such file or directory
```

Comment (pierDipi) [reply test/eventing-kafka.bash:40]: cc @mgencur @matzew I need a couple of fresh eyes to catch what's wrong

Comment (matzew) [reply test/eventing-kafka.bash:40]: OK, I will take a look :+1:  

Sorry but I have not seen the comment before :sweat_smile:

File: test/eventing-kafka.bash [hunk 2]
```diff
@@ -29,3 +29,29 @@ function upstream_knative_eventing_kafka_e2e {
   logger.info 'Starting eventing-kafka tests'
   run_e2e_tests
 }
+
+function upstream_knative_eventing_kafka_broker_e2e {
+  logger.info 'Setting Kafka as default broker class'
+
+  local root_dir
+  root_dir=$(dirname "$0")/..
```
Comment (matzew) [inline test/eventing-kafka.bash:37]: Are you sure this is the right ?

Comment (mgencur) [reply test/eventing-kafka.bash:37]: I think this should be:
```
rootdir=""$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")"
```

Comment (pierDipi) [reply test/eventing-kafka.bash:37]:
```suggestion
  root_dir="$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")"
```

**Comment by mgencur** [comment]
Regarding the lint error, we often use this to disable it throughout the s-o repo:
# shellcheck disable=SC1091,SC1090

File: test/eventing-kafka.bash (hunk 2 above)
Comment (matzew) [reply test/eventing-kafka.bash]: That looks beautiful :joy:

File: test/eventing-kafka.bash [hunk 3]
```diff
@@ -34,7 +34,7 @@ function upstream_knative_eventing_kafka_broker_e2e {
   logger.info 'Setting Kafka as default broker class'
 
   local root_dir
-  root_dir=$(dirname "$0")/..
+  root_dir="$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")"
```
Comment (matzew) [inline test/eventing-kafka.bash:37]: what a lovely script :sweat_smile:

File: test/eventing-kafka.bash (hunk 2 above)
Comment (mgencur) [reply test/eventing-kafka.bash]: Right ?!? :smile:  But it should be bullet-proof cos it doesn't depend on where you're running the tests from.

**Comment by matzew** [comment]
```

Error from server (UnsupportedMediaType): the body of the request was in an unknown format - accepted media types include: application/json-patch+json, application/merge-patch+json, application/apply-patch+yaml
13:50:21.938 ERROR:   🚨 Error (code: 1) occurred at /go/src/github.com/openshift-knative/serverless-operator/test/eventing-kafka.bash:40, with command: oc patch knativeeventing -n knative-eventing knative-eventing --patch-file "${root_dir}/test/config/eventing/kafka-broker-default-patch.yaml"
```

**Comment by pierDipi** [comment]
I think I found the problem for that error: https://github.com/openshift-knative/serverless-operator/pull/1674/commits/0e69894117f77ef23d36b912bd4177a79108bb01

**Comment by matzew** [comment]
```
error: build error: error building at STEP "RUN yum install -y kubectl httpd-tools": error while running runtime: exit status 1
```

**Comment by pierDipi** [comment]
A pre-refactoring https://github.com/openshift-knative/serverless-operator/pull/1731 (or commonly know as prefactoring)

**Comment by pierDipi** [comment]
CI job: https://github.com/openshift/release/pull/32603

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/hold
(just for the separated tests)

**Comment by pierDipi** [comment]
Failed
/hold

/test 4.11-upstream-e2e-kafka-aws-ocp-411

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/hold
for remaining test

**Pull request merged**